packages/widget_toolkit_pin/lib/src/lib_pin_code_with_biometrics/ui_components/pin_code_keyboard.dart

```diff
@@ -0,0 +1,149 @@
+/// in the two customizable button options on the right direction. To define them use
+/// [deleteKeyButton] and/or [bottomRightKeyboardButton]. The right button by default
+/// can be is an auto submit button, delete button and biometrics button.
+/// When the widget is loaded for the first time, on the bottom right of the
+/// page there is no button. At this moment the biometrics for the app are still
+/// not enabled. After at least 1 number has been selected the delete button
+/// shows up. When the length of the input has reached the pin code length, the
+/// button icon disappears. The pin code is encrypted and stored in the
+/// local device secure storage. Then, there is an auto submit for the selected
+/// pin code to the backend for verification. When the pin code is submitted,
+/// the widget should present to the user that the pin verification is in
+/// progress with loading animation. Over the masked pin a Shimmer will be
+/// presented and buttons will change their appearance. The pin code from the
+/// input is auto submitted once its length reaches the returned value from
+/// [PinCodeService.getPinLength()], which should return a value less than 10.
+/// After the pin has been saved
+/// successfully in the secure storage, the biometrics icon appear on the bottom
+/// right. When you press it, it triggers enabling of the biometrics event. The
+/// local authentication from the local_auth package is triggered. The user is
+/// asked, if he/she wants to allow the app to use biometrics authentication.
+/// When he/she clicks ok, the biometrics authentication is triggered. When it
+/// is successful, on the screen is displayed a message that the biometrics are
+/// enabled. The next time when the app is restarted, because the pin code
+/// will be stored in the device secure storage, the biometrics authentication
+/// will be automatically triggered and the biometrics icon will be displayed
+/// on the bottom right. When you press it every time it will trigger the
+/// biometric authentication.
+///
+/// For more information on how the widgets work check the functional specification
+/// section in the README.md file.
+class PinCodeKeyboard extends StatelessWidget {
+  const PinCodeKeyboard({
+    required this.pinCodeService,
+    required this.biometricsLocalDataSource,
+    required this.translateError,
+    this.mapMessageToString,
+    this.isAuthenticatedWithBiometrics,
+    this.isPinCodeVerified,
+    this.deleteKeyButton,
+    this.bottomRightKeyboardButton,
+    this.localizedReason,
+    this.addDependencies = true,
+    this.onError,
+    Key? key,
+  }) : super(key: key);
+
+  /// Handle the translation of the error from the errors stream
+  final String Function(Object error) translateError;
+
+  /// [mapMessageToString] will be used to translate the [BiometricsMessage]
+  /// to human readable text and will be used into the default notification
+  final String Function(BiometricsMessage message)? mapMessageToString;
```

Comment:
Change this name to something more concrete. The name is to generic but its usage is specific. For ex. `mapBiometricMessageToString`

Reply:
I have set the name to `mapBiometricMessageToString`